executor/src/test/java/org/k8loud/executor/action/kubernetes/HorizontalScalingActionTest.java

```diff
@@ -3,151 +3,40 @@
-                .withName("depl1")
-                .withNamespace("test")
-                .withResourceVersion("1")
-                .endMetadata()
-                .withNewSpec()
-                .withReplicas(3)
-                .endSpec()
-                .withNewStatus()
-                .withReplicas(2)
-                .endStatus()
-                .build();
-        client.resource(depl)
-                .create();
-
-        //when
-        Action action = new HorizontalScalingAction(params, kubernetesService);
-        ExecutionRS rs = action.perform();
-        Deployment depl1 = client.apps()
-                .deployments()
-                .withName(params.getRequiredParam("resourceName"))
-                .get();
-
-        //then
-        assertEquals(ExecutionExitCode.OK, rs.getExitCode());
-        assertNotNull(depl1);
-        assertNotNull(depl1.getSpec());
-        assertNotNull(depl1.getStatus());
-        assertEquals(Integer.parseInt(params.getRequiredParam("replicas")), depl1.getSpec()
-                .getReplicas()
-                .intValue());
-    }
-
-    public static Stream<Params> provideReplicasetScalingParams() {
-        return Stream.of(new Params(
-                        Map.of("resourceType", "ReplicaSet", "resourceName", "repl1", "namespace", "test", "replicas"
-                                , "3")),
-                new Params(
-                        Map.of("resourceType", "ReplicaSet", "resourceName", "repl1", "namespace", "test", "replicas",
-                                "1")), new Params(
-                        Map.of("resourceType", "ReplicaSet", "resourceName", "repl1", "namespace", "test", "replicas",
-                                "2")));
-    }
-
-    @ParameterizedTest
-    @MethodSource("provideReplicasetScalingParams")
-    void testScalingReplicaSet(Params params) throws ActionException {
-
-        // given
-        ReplicaSet rss = new ReplicaSetBuilder().withNewMetadata()
-                .withName("repl1")
-                .withNamespace("test")
-                .withResourceVersion("1")
-                .endMetadata()
-                .withNewSpec()
-                .withReplicas(2)
-                .endSpec()
-                .withNewStatus()
-                .withReplicas(2)
-                .endStatus()
-                .build();
-        client.resource(rss)
-                .create();
-
+    @MethodSource
+    void testIfKubernetesServiceIsCalled(Params params) throws ActionException, KubernetesException {
         // when
         Action action = new HorizontalScalingAction(params, kubernetesService);
         ExecutionRS rs = action.perform();
-        ReplicaSet repl = client.apps()
-                .replicaSets()
-                .withName(params.getRequiredParam("resourceName"))
-                .get();
 
         // then
+        verify(kubernetesService, times(1)).scaleHorizontally(params.getRequiredParam(NAMESPACE_KEY),
```

Comment:
Also change to `NAMESPACE` etc, not `params.getRequiredParam` and add assert for message check

Reply:
The test would break if I changed to `NAMESPACE` and someone passed a different namespace value in params.
I'll get rid of the parametrized test though and change like you suggested since the second test case doesn't check anything more than the first one

Reply:
I guess by `message` you mean `ExecutionRS.result` which depends on the KubernetesService's result which is out of the scope of this test

Reply:
Added result checks in KubernetesService's tests